src/components/Post/PostDetailComponent.jsx

```diff
@@ -1,16 +1,135 @@
 import React, { Component } from 'react';
-import { Segment } from 'semantic-ui-react';
+import { Segment, Image } from 'semantic-ui-react';
+import { withRouter, Link } from 'react-router-dom';
+import GoogleMapReact from 'google-map-react';
+
+import MapMarker from '../Home/MapMarkers';
 
 class NewPost extends Component {
+  state = {
+    zoom:15,
+    loading: false
```

Comment:
Seems like the `post` state is not initialized here. If I were you, any state that's going to be used here, I would initialized them before the component `constructor` is called for the following reasons:

1. If someone is reviewing your code wants to backtrack how data is being rendered. It would be nice where the state is coming from.
2. When the `render()` method is invoked, it's using the appropriate state to render some JSX, but it's going to throw an `undefined` error if the state is not initialized.